### PR TITLE
Fix disabled tslint no-unused-expression

### LINKED ISSUE #616: Fix disabled no-unused-expression ts rule
- Split new branch from ts branch

### PR DESCRIPTION
**Pull request by wsmd**
It's exciting to see the project migrating to TS! 🎉 

This should resolve #616 

**Before**

```
yarn run tslint 'src/**/*.{ts,tsx}' | grep 'unused expression'

ERROR: src/modules/common/styles/global-styles.ts[4, 1]: unused expression, expected an assignment or function call
ERROR: src/modules/common/utils/searchers.ts[15, 7]: unused expression, expected an assignment or function call
ERROR: src/modules/common/utils/searchers.ts[32, 7]: unused expression, expected an assignment or function call
ERROR: src/modules/common/utils/searchers.ts[46, 7]: unused expression, expected an assignment or function call
ERROR: src/modules/deals/components/deal/editForm/Sidebar.tsx[40, 7]: unused expression, expected an assignment or function call
ERROR: src/modules/deals/components/product/ProductItemForm.tsx[68, 5]: unused expression, expected an assignment or function call
ERROR: src/modules/deals/components/product/ProductItemForm.tsx[82, 7]: unused expression, expected an assignment or function call
ERROR: src/modules/inbox/components/conversationDetail/workarea/RespondBox.tsx[164, 9]: unused expression, expected an assignment or function call
ERROR: src/modules/inbox/components/conversationDetail/workarea/RespondBox.tsx[168, 9]: unused expression, expected an assignment or function call
ERROR: src/modules/inbox/containers/conversationDetail/ConversationDetail.tsx[33, 7]: unused expression, expected an assignment or function call
ERROR: src/modules/inbox/containers/conversationDetail/ConversationDetail.tsx[34, 7]: unused expression, expected an assignment or function call
ERROR: src/modules/inbox/containers/conversationDetail/WorkArea.tsx[177, 9]: unused expression, expected an assignment or function call
ERROR: src/modules/inbox/containers/conversationDetail/WorkArea.tsx[180, 9]: unused expression, expected an assignment or function call
ERROR: src/modules/settings/brands/components/ChooseBrand.tsx[40, 5]: unused expression, expected an assignment or function call
ERROR: src/modules/settings/integrations/components/common/ManageIntegrations.tsx[101, 5]: unused expression, expected an assignment or function call
ERROR: src/modules/settings/integrations/components/common/ManageIntegrations.tsx[102, 5]: unused expression, expected an assignment or function call
ERROR: src/modules/settings/integrations/components/common/ManageIntegrations.tsx[103, 5]: unused expression, expected an assignment or function call
ERROR: src/modules/settings/integrations/components/common/ManageIntegrations.tsx[112, 5]: unused expression, expected an assignment or function call
ERROR: src/modules/settings/integrations/components/common/ManageIntegrations.tsx[113, 5]: unused expression, expected an assignment or function call
ERROR: src/modules/settings/integrations/components/common/ManageIntegrations.tsx[114, 5]: unused expression, expected an assignment or function call
ERROR: src/modules/settings/profile/components/ChangePassword.tsx[39, 5]: unused expression, expected an assignment or function call
```

**After**

No errors! 🎉

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "no-var-requires": false,
     "object-literal-sort-keys": false,
     "curly": false,
-    "no-unused-expression": false,
+    "no-unused-expression": [true, "allow-tagged-template"],
```
Comment (wsmd) [inline tslint.json]: I had to add this extra configuration to allow using tagged template in order to get around this:

```
ERROR: src/modules/common/styles/global-styles.ts[4, 1]: unused expression, expected an assignment or function call
```

https://github.com/erxes/erxes/blob/0cbbf446c97e678399a84dfd1b207c0d70f17eba/src/modules/common/styles/global-styles.ts#L4-L6

More on this: https://palantir.github.io/tslint/rules/no-unused-expression/

Comment (wsmd) [reply tslint.json]: Alternatively, I could remove this configuration entirely and use something like `tslint:disable-next-line` in the file itslef. What do you think?

**Comment by batamar** [comment]
@wsmd Thanks for the pull request. There are some conflicting files.

By the way, There are some complier errors in #615. Did you run yarn start command or did you just check compiler errors in your editor ?

**Comment by batamar** [comment]
@wsmd Thanks again. Nice job 👍

**Pull request merged**